### ISSUE TITLE
Not able to set percentage to the width of components

### ISSUE THREAD

**Issue by srikanthankem** [JarekToro/responsive-layout#26]
In the alpha version i was able to set percentage as width to the components and its working fine when i upgraded to new version the percentage as width is working but the responsiveness goes away.

**Comment by srikanthankem** [comment]
Sorry percentage as width is not working and i was able to put pixels as width and its working but the responsiveness gone

**Comment by JarekToro** [comment]
What happens when you use percentage as width on the components? Does it have no width? Does it overflow the ResponsiveColumn. And how do you set the width? Do you use setWidth or set SizeFull?

**Comment by srikanthankem** [comment]
I created the parent layout and my parent layout is asusal responsivelayout and i added row to the layout and added columns to the row .This component will be my horizontal layout and when i do setsizefull  to the layout its not expanding to the screen but limits to the width of the columns and components with in the columns.I also tried to setwidth 100% and still its not working.
I usually created layout then add row to the layout and then add columns to the row thats how i constructed. 
Will i have to do the same way as you constructed  the layout means layout.addrow while creating itself .I dont think so because previous version works fine.I event created custom components like grid accordion panel using the previous responsive framework

**Comment by JarekToro** [comment]
To make sure I'm correct this is how you have it laid out

ResponsiveLayout
>>ResponsiveRow
    >>ResponsiveColumn
        >>HorizontalLayout

**Comment by srikanthankem** [comment]
This is the hierarchy i followed and in the Responsive Column i have vaadin components like panel and with in panel i have label

ResponsiveLayout
-- ResponsiveRow
    -- ResponsiveColumn
        --Panel
           --Label

**Comment by srikanthankem** [comment]
ResponsiveLayout
 ---)) ResponsiveRow
       --- ))ResponsiveColumn
              --)) Panel
                   --)) Label

**Comment by JarekToro** [comment]
The Panel is the component set to 100%

**Comment by JarekToro** [comment]
I'm on my phone so I closed it on accident

**Comment by srikanthankem** [comment]
The layout is divided in to 4 columns and each column has panel when i set the panel to setfullSize its not expanding full to the screen it expanded up to the label text size which is inside the panel.
The same code work beautifully in the alpha version. I am not able to judge the new behaviour some how

**Comment by JarekToro** [comment]
Ok I understand, let me see what I can find

**Comment by JarekToro** [comment]
ok if this is the case

```java
ResponsiveLayout  root = new ResponsiveLayout();

ResponsiveRow row = new ResponsiveRow();

root.addRow(row);

//Column that takes up 3/12 of screen on all mobile sizes.
ResponsiveColumn column = new ResponsiveColumn(3,3,3,3);

row.addColumn(column);


Panel panel = new Panel("test");
panel.setSizeFull();

column.setComponent(panel);

```

**Using setSizeFull() should make the panel only take up all the space of its parent which in this case is the column which is currently taking up 3/12 of the page. So in this case it is doing its job. Im thinking in alpha maybe that was a bug in the layout mechanism.**


------
### Possibile Alternatives for this behavior

If you want the panel to take up 100% of the screen itself, you can either set the column that it is in to 12.

```java
column.addRule(12, displaySize.xs);
column.addRule(12, displaySize.sm)
column.addRule(12, displaySize.md)
column.addRule(12, displaySize.lg)
```  
which will cause the column to expand to fill the Row completely.

or add you can add the panel to the ResponsiveLayout temporary and call panel.setSizeFull();

then move it back to the column whenever.


### References 
-----
Sizable class and the SetSizeFull Function.
https://vaadin.com/api/7.7.6/com/vaadin/server/Sizeable.html
https://vaadin.com/api/7.7.6/com/vaadin/server/Sizeable.html#setSizeFull--

**Comment by srikanthankem** [comment]
The solution you suggested is useful for verticallayout style position not for horizontal. But some how  the CSS before(alpha ) version is more flexible to add components with in components and the responsiveness is beautiful and the current version is not much supportive to do.
The newest version is taking more time and responsiveness in old version is very good when compared to new one as i actually worked on these two versions

**Comment by JarekToro** [comment]
I can't say that I have experienced any of these issues. I do use ResponsiveLayout in my projects and of course all the versions in between I have used. Maybe I am not understanding the issue.

Would it be possible for you to create a simple UI.class that works how you want in the old version and doesn't in the new version.  Then I can see first hand the issue your having. This would help tremendously.
